tests/integration/pilot/common/routing.go

```diff
@@ -3600,8 +3594,11 @@ spec:
 				address += "&server=" + tt.server
 			}
 			var checker echo.Checker = func(result echo.CallResult, _ error) error {
+				if len(result.Responses) == 0 {
+					return fmt.Errorf("no responses")
+				}
 				for _, r := range result.Responses {
-					if !reflect.DeepEqual(r.Body(), tt.expected) {
+					if !sets.New(r.Body()...).Equals(sets.New(tt.expected...)) {
```

Comment:
Two bug fixes here unrelated to the change:
1. Fail if we get no response. Right now we succeed! So its a race, we basically skip this test often
2. Compare unordered. Right now we do in-order, so it may take a few attempts for the random order to match up